src/sentry/event_manager.py

```diff
@@ -513,16 +523,14 @@ def save(self, project_id, raw=False, assume_normalized=False):
 
         # Check to make sure we're not about to do a bunch of work that's
         # already been done if we've processed an event with this ID. (This
-        # isn't a perfect solution -- this doesn't handle ``EventMapping`` and
-        # there's a race condition between here and when the event is actually
-        # saved, but it's an improvement. See GH-7677.)
-        try:
-            event = Event.objects.get(project_id=project.id, event_id=data["event_id"])
-        except Event.DoesNotExist:
-            pass
-        else:
+        # isn't a perfect solution -- there's a race condition between here and
+        # when the event is actually saved, but it's an improvement. See GH-7677.)
+        event = self._get_event(project_id, data["event_id"])
```

Comment:
When we get here we have already excluded duplicate event ids that were in memcache, is that right ?

Reply:
Yep, that's right